codex/stores/repostore.nim

```diff
@@ -222,6 +243,41 @@ method listBlocks*(
   iter.next = next
   return success iter
 
+proc createBlockExpirationQuery(maxNumber: int, offset: int): ?!Query =
+  let queryString = ? (BlocksTtlKey / "*")
+  let queryKey = ? Key.init(queryString)
+  success Query.init(queryKey, offset = offset, limit = maxNumber)
+
+method getBlockExpirations*(self: RepoStore, maxNumber: int, offset: int): Future[?!BlockExpirationIter] {.async, base.} =
```

Comment:
In the current arrangement, the maintenance module does not interact with any datastores, only with the repoStore.
If we move getBlockExpirations to maintenance, then it will need to know about the metadata Sqllite store, and both it and the repostore will be using it. I figured I'd keep the knowledge of the datastore and how to format keys for it in one place, but I do agree getBlockExpirations feels a bit out of place here.
Suppose I could move the creating and retrieving of the expiration metadata both to the maintenance module? Then repostore uses the metadata store for the quote key, and maintenance uses it for expiration.... Do you like this plan? [Y/N]